include/engine/guidance/route_step.hpp

```diff
@@ -144,6 +146,14 @@ inline RouteStep &RouteStep::AddInFront(const RouteStep &preceeding_step)
     return *this;
 }
 
+// checks whether we can actually merge two route steps:
+// They neeed to have the same mode, because a route step
+// can only have one mode.
+inline bool RouteStep::IsCompatible(const RouteStep &other) const
+{
+    return mode == other.mode;
+}
```

Comment:
Hm I'm a bit out of the loop here but I thought we had this exact function already.
Can't see it on the route step, so this should be fine for sure.